Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/IdentityRepository.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Domain.Entities;
+using Backbone.Modules.Devices.Infrastructure.Persistence.Database;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.Database;
+using Enmeshed.BuildingBlocks.Application.Extensions;
+using Enmeshed.BuildingBlocks.Application.Pagination;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backbone.Modules.Devices.Infrastructure.Persistence.Repository;
+public class IdentityRepository : IIdentityRepository
+{
+    private readonly DbSet<Identity> _identityDbSet;
+    private readonly IQueryable<Identity> _readonlyIdentityDbSet;
+
+    public IdentityRepository(DevicesDbContext dbContext)
+    {
+        _identityDbSet = dbContext.Identities;
+        _readonlyIdentityDbSet = dbContext.Identities.AsNoTracking();
```

Comment:
You didn't test this, did you? Because this code should throw a NullReferenceException when accessing complex child properties, like "Devices`. You have to tell EF Core to actually read them from the database.

Reply:
Well spotted. No NullReferenceException, but the counts were returning zero.